File: cmd/client/minio.go
package main

import (
	"context"
	"fmt"
	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
	"log"
	"os"
	"path/filepath"
)

type MinioClient struct {
	minioClient *minio.Client
}

// NewMinioClient to initialize MinIO client
func NewMinioClient(minioEndpoint, minioAccessKeyID, minioSecretAccessKey string, minioUseSSL bool) *MinioClient {
	minioClient, err := minio.New(minioEndpoint, &minio.Options{
		Creds:  credentials.NewStaticV4(minioAccessKeyID, minioSecretAccessKey, ""),
		Secure: minioUseSSL,
	})
	if err != nil {
		log.Fatalln(err)
	}

	return &MinioClient{
		minioClient: minioClient,
	}
}

// CreateBucket to create bucket
func (m *MinioClient) CreateBucket(bucketName string) {
	err := m.minioClient.MakeBucket(context.Background(), bucketName, minio.MakeBucketOptions{})
	if err != nil {
		log.Fatalln(err)
	}

	fmt.Printf("Successfully created bucket %s\n", bucketName)
}

// UploadFile to upload the file into minio
func (m *MinioClient) UploadFile(bucketName, filePath string) {
	fileName := filepath.Base(filePath)

	if !fileExists(fileName) {
		log.Fatalf(fmt.Sprintf("File %s does not exist", fileName))
	}

	_, err := m.minioClient.FPutObject(context.Background(), bucketName, fileName, filePath, minio.PutObjectOptions{})
	if err != nil {
		log.Fatalln(err)
	}

	fmt.Printf("Successfully uploaded %s to %s/%s\n", filePath, bucketName, fileName)
}

// fileExists checks if the specified file exists.
func fileExists(filePath string) bool {
	_, err := os.Stat(filePath)
	if os.IsNotExist(err) {
		return false
	}
	return err == nil
}

File: ocr/src/ocr/lib/clients/minio.py
from minio import Minio
from minio.error import S3Error
from .utils.env_util import Environment


class MinioClient:
    def __init__(self, minio_endpoint: str, access_key: str, secret_key: str, bucket_name: str, secure: bool = False):
        self.client = Minio(
            minio_endpoint,
            access_key=access_key,
            secret_key=secret_key,
            secure=False
        )
        self.bucket_name = bucket_name

        if not self.client.bucket_exists(self.bucket_name):
            self.client.make_bucket(self.bucket_name)

    def upload_file(self, file_path, object_name):
        try:
            self.client.fput_object(
                self.bucket_name, object_name, file_path
            )
            print(f"File uploaded to MinIO: {object_name}")
        except S3Error as e:
            print(f"Failed to upload file: {e}")
